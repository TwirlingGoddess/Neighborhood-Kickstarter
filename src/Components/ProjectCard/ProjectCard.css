.Card {  
  display: inline-block;
  position: relative;
  height: 20rem;
  width: 45rem;
  background-image: linear-gradient(to top, #09203f 0%, #537895 100%);
  margin: 1rem 3rem;
}