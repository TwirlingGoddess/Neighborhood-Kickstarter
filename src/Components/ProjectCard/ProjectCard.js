import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './ProjectCard.css';
import PropTypes from 'prop-types';

class ProjectCard extends Component {
  constructor() {
    super();
  }
  
  viewComments = (project) => {
    this.props.selectProject(project);
  }

  render() {
    const { title, neighbor, contact, description, photo, resources, first_name, last_name, email } = this.props;
    if (title) {
      return (
        <div className='Card'>
          <img className='project-image' src={photo} alt='pciture of the project'/>
          <h2>project: {title}</h2>
          <h3>neighbor: {neighbor ? neighbor : first_name + ' ' + last_name}</h3>
          <h4>contact: {contact ? contact : email}</h4>
          <h5>description: {description}</h5>
          <h5>resources: </h5>
          {resources.map((resource, index) => {
            return <h5 key={index}>{resource.name}</h5>;
          })}
          <NavLink onClick={() => this.viewComments(this.props)} to='/Contributions'>Comments</NavLink>
        </div>
      );
    } else {
      return (
        <div>
          <h1>No Projects for this neighborhood</h1>
        </div>
      );
    }
  }
}

ProjectCard.propTypes = {
  selectProject: PropTypes.func,
  title: PropTypes.string,
  neighbor: PropTypes.string,
  contact: PropTypes.string,
  description: PropTypes.string,
  photo: PropTypes.string,
  resources: PropTypes.array,
  first_name: PropTypes.string,
  last_name: PropTypes.string,
  email: PropTypes.string
};

export default ProjectCard;