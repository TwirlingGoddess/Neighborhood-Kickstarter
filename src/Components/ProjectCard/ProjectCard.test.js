import React from 'react';
import ProjectCard from './ProjectCard';
import { shallow } from 'enzyme';
import { NavLink } from 'react-router-dom';

describe('ProjectCard', () => {
  let wrapper;
  it('should match the snapshot', () => {
    wrapper = shallow(<ProjectCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call selectProject when viewComments is called', () => {
    let mockSelectProject = jest.fn();
    wrapper = shallow(<ProjectCard selectProject={mockSelectProject}/>);

    wrapper.instance().viewComments();

    expect(mockSelectProject).toHaveBeenCalled();
  });

  it('should only render a view comments button if there is a current user', () => {
    let mockCurrentUser = {}
    wrapper = shallow(<ProjectCard currentUser={mockCurrentUser}/>)

    expect(wrapper.find(NavLink)).toHaveLength(0)
  })
}); 