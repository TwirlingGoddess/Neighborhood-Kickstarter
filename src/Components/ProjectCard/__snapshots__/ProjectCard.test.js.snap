// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectCard
    resources={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="uploaded view of the project"
          className="project-image"
        />,
        <h2>
          project: 
        </h2>,
        <h3>
          neighbor: 
          undefined undefined
        </h3>,
        <h4>
          contact: 
        </h4>,
        <h5>
          description: 
        </h5>,
        <h5>
          resources: 
        </h5>,
        Array [],
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="comments-button"
          onClick={[Function]}
          to="/Contributions"
        >
          Comments
        </NavLink>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "uploaded view of the project",
          "className": "project-image",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "project: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "project: ",
          undefined,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "neighbor: ",
            "undefined undefined",
          ],
        },
        "ref": null,
        "rendered": Array [
          "neighbor: ",
          "undefined undefined",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "contact: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "contact: ",
          undefined,
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "description: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "description: ",
          undefined,
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "resources: ",
        },
        "ref": null,
        "rendered": "resources: ",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Comments",
          "className": "comments-button",
          "onClick": [Function],
          "to": "/Contributions",
        },
        "ref": null,
        "rendered": "Comments",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="uploaded view of the project"
            className="project-image"
          />,
          <h2>
            project: 
          </h2>,
          <h3>
            neighbor: 
            undefined undefined
          </h3>,
          <h4>
            contact: 
          </h4>,
          <h5>
            description: 
          </h5>,
          <h5>
            resources: 
          </h5>,
          Array [],
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="comments-button"
            onClick={[Function]}
            to="/Contributions"
          >
            Comments
          </NavLink>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "uploaded view of the project",
            "className": "project-image",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "project: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "project: ",
            undefined,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "neighbor: ",
              "undefined undefined",
            ],
          },
          "ref": null,
          "rendered": Array [
            "neighbor: ",
            "undefined undefined",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "contact: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "contact: ",
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "description: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "description: ",
            undefined,
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "resources: ",
          },
          "ref": null,
          "rendered": "resources: ",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Comments",
            "className": "comments-button",
            "onClick": [Function],
            "to": "/Contributions",
          },
          "ref": null,
          "rendered": "Comments",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
